packages/hash/frontend/src/pages/[account-slug]/entities/[entity-id].page/entity-editor/properties-section/property-table/cells/value-cell/value-cell-editor/array-editor.tsx

```diff
@@ -0,0 +1,183 @@
+import { Box } from "@mui/material";
+import produce from "immer";
+import { useMemo, useRef, useState } from "react";
+import {
+  closestCenter,
+  DndContext,
+  KeyboardSensor,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from "@dnd-kit/core";
+import {
+  arrayMove,
+  SortableContext,
+  sortableKeyboardCoordinates,
+  verticalListSortingStrategy,
+} from "@dnd-kit/sortable";
+import { ValueCellEditorComponent } from "../types";
+import { SortableItem } from "./array-editor/types";
+import { SortableRow } from "./array-editor/sortable-row";
+import { AddAnotherButton } from "./array-editor/add-another-button";
+import { InlineTextEditor } from "./array-editor/inline-text-editor";
+
+const NEW_ROW_KEY = "new";
+
+export const ArrayEditor: ValueCellEditorComponent = ({
+  value: cell,
+  onChange,
+}) => {
+  const scrollableContainer = useRef<HTMLDivElement>(null);
+
+  const items = useMemo(() => {
+    const propertyVal = cell.data.property.value;
+    const values = Array.isArray(propertyVal) ? propertyVal : [propertyVal];
+
+    const itemsArray: SortableItem[] = values.map((value, index) => ({
+      index,
+      id: `${index}_${String(value)}`,
+      value,
+    }));
+
+    return itemsArray;
+  }, [cell]);
+
+  const [input, setInput] = useState("");
+  const [selectedRow, setSelectedRow] = useState("");
+  const [editingRow, setEditingRow] = useState(items.length ? "" : NEW_ROW_KEY);
+
+  const addItem = (text: string) => {
+    const newCell = produce(cell, (draftCell) => {
+      draftCell.data.property.value = [
+        ...items.map(({ value }) => value),
+        text,
+      ];
+    });
+    onChange(newCell);
+  };
+
+  const removeItem = (index: number) => {
+    const newCell = produce(cell, (draftCell) => {
+      draftCell.data.property.value = items
+        .filter((_, index2) => index !== index2)
+        .map(({ value }) => value);
```

Comment:
I think this makes it a bit clearer
```suggestion
  const removeItem = (indexToRemove: number) => {
    const newCell = produce(cell, (draftCell) => {
      draftCell.data.property.value = items
        .filter((_, index) => indexToRemove !== index)
        .map(({ value }) => value);
```